File: solutions/typescript/rna-transcription/1/rna-transcription.ts
export function toRna(strand: string): string {
  return strand
    .split("")
    .map((n) => {
      if (!dict[n]) throw new Error("Invalid input DNA.");
      return dict[n];
    })
    .join("");
}

const dict: Record<string, string> = {
  G: "C",
  C: "G",
  T: "A",
  A: "U",
};

File: solutions/ruby/rna-transcription/1/rna_transcription.rb
module Complement
  COMPLEMENTS = {
    "G" => "C",
    "C" => 'G',
    'T' => 'A',
    'A' => 'U'
  }
  
  def self.of_dna(strand)
    return "" if strand == ""
    strand.each_char.map {|c| COMPLEMENTS[c] }.join
  end
end